control-base/src/main/java/fi/nls/oskari/control/data/GetCSWDataHandler.java

```diff
@@ -80,10 +81,13 @@ public void handleAction(ActionParameters params) throws ActionException {
         final String uuid = params.getRequiredParam(UUID_PARAM);
         // TODO use default lang if not found?
         final String lang = params.getRequiredParam(LANG_PARAM);
+        
+        String url = params.getHttpParam(METADATA_URL_PARAM, baseUrl);
```

Comment:
This allows anyone to make calls to arbitrary URLs from the server. Let's just send the layer id and get the uuid and optional url from the layer instead.